Libraries/LibWeb/CSS/StyleValues/ShorthandStyleValue.cpp

```diff
@@ -348,23 +349,68 @@ String ShorthandStyleValue::to_string(SerializationMode mode) const
         if (!first_is_one_of(font_variant_string, "normal"sv, "small-caps"sv) && !CSS::is_css_wide_keyword(font_variant_string)) {
             return {};
         }
+
         // <font-width-css3> = normal | ultra-condensed | extra-condensed | condensed | semi-condensed | semi-expanded | expanded | extra-expanded | ultra-expanded
-        switch (font_width->to_keyword()) {
-        case Keyword::Initial:
-        case Keyword::Normal:
-        case Keyword::UltraCondensed:
-        case Keyword::ExtraCondensed:
-        case Keyword::Condensed:
-        case Keyword::SemiCondensed:
-        case Keyword::SemiExpanded:
-        case Keyword::Expanded:
-        case Keyword::ExtraExpanded:
-        case Keyword::UltraExpanded:
-            break;
-        default:
-            if (!font_width->is_css_wide_keyword())
+        auto font_width_as_keyword = [&]() -> Optional<Keyword> {
+            if (first_is_one_of(font_width->to_keyword(), Keyword::Initial, Keyword::Normal, Keyword::UltraCondensed, Keyword::ExtraCondensed, Keyword::Condensed, Keyword::SemiCondensed, Keyword::SemiExpanded, Keyword::Expanded, Keyword::ExtraExpanded, Keyword::UltraExpanded))
+                return font_width->to_keyword();
+
+            Optional<double> font_width_as_percentage;
+
+            if (font_width->is_percentage())
+                font_width_as_percentage = font_width->as_percentage().raw_value();
+            else if (font_width->is_calculated())
+                // NOTE: We don't pass a length resolution context but that's fine because either:
+                //  - We are working with declarations in which case relative units can't be mapped so their mere
+                //    presence means we can't serialize this font shorthand
+                //  - We are working with computed values in which case we would have already converted any
+                //    CalculatedStyleValues values to normal PercentageStyleValues
+                font_width_as_percentage = font_width->as_calculated().resolve_percentage({}).map([](auto percentage) { return percentage.value(); });
+
+            if (!font_width_as_percentage.has_value())
                 return {};
-        }
+
+            // ultra-condensed 50%
+            if (*font_width_as_percentage == 50)
```

Comment:
Nit: You don't need the `*` here and below: `Optional<T>` is directly comparable with `T`.